pkg/apis/metalkube/v1alpha1/baremetalhost_types.go

```diff
@@ -342,6 +342,10 @@ func (host *BareMetalHost) CredentialsNeedValidation(currentSecret corev1.Secret
 // status and returns true when more work is needed or false
 // otherwise.
 func (host *BareMetalHost) NeedsProvisioning() bool {
+	if !host.Spec.Online {
```

Comment:
It would be helpful to describe this in the docs for the host spec. Mentioning this on the `Online` field would be a good idea, but it feels like we're at a point where it would be good to summarize what things a user needs to do to their Spec in order to initiate provisioning.

It also might help to express in the host status that it's not provisioning because the host is powered off.

Reply:
I've added the docs.
I'm not sure about updating the host status to say why it isn't doing something that isn't being requested, but we can talk more about that.

Reply:
Sounds good. We can work on that status later. Perhaps we should even do validation and prevent someone from setting the image while power is off. Generally, I'm just looking at it as an invalid state; the user has asked us to ensure that the host is provisioned while the power is off. We can't do both of those.